services/src/main/java/org/exoplatform/addons/gamification/service/setting/rule/impl/RuleRegistryImpl.java

```diff
@@ -77,10 +77,9 @@ public void start() {
 
             for (RuleConfig rule : ruleMap.values()) {
                 RuleDTO ruleDTO = ruleService.findRuleByTitle(GAMIFICATION_DEFAULT_DATA_PREFIX+rule.getTitle());
-                if (ruleDTO == null) {
-                    store(rule);
+                if (ruleDTO == null || !(ruleDTO.getEvent().equals(rule.getEvent()))) {
```

Comment:
why don't you check if the title is not the same ?

Reply:
it's the  same but if i will check with title I must concatenate the title with this constant GAMIFICATION_DEFAULT_DATA_PREFIX so the condition will be :
if(ruleDTO == null || !(ruleDTO.getTitle().equals(GAMIFICATION_DEFAULT_DATA_PREFIX+rule.getTitle()))